`libc::abort` prevents `wasm` build

Building for wasm currently fails because `libc::abort` is not available on this target.

```
$ rustup target add wasm32-unknown-unknown
$ cargo build --target wasm32-unknown-unknown
error[E0425]: cannot find function `abort` in module `libc`=======>    ] 17/18: crossbeam-channel                                                                                                                
   --> src/internal/channel.rs:268:28
    |
268 |             unsafe { libc::abort() }
    |                            ^^^^^ not found in `libc`
help: possible candidates are found in other modules, you can import them into scope
    |
3   | use std::intrinsics::abort;
    |
3   | use std::process::abort;
    |

error[E0425]: cannot find function `abort` in module `libc`
  --> src/internal/utils.rs:61:24
   |
61 |         unsafe { libc::abort() }
   |                        ^^^^^ not found in `libc`
help: possible candidates are found in other modules, you can import them into scope
   |
3  | use std::intrinsics::abort;
   |
3  | use std::process::abort;
   |

error: aborting due to 2 previous errors
```

We're pretty close to at least compiling! Do you think it would be possible to use an alternative to `libc::abort` in these cases?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,6 @@ categories = ["algorithms", "concurrency", "data-structures"]
 [dependencies]
 crossbeam-epoch = "0.5.0"
 crossbeam-utils = "0.4.0"
-libc = "0.2.42"
 parking_lot = "0.5"
 rand = "0.5.3"
 smallvec = "0.6.2"

--- a/src/internal/channel.rs
+++ b/src/internal/channel.rs
@@ -6,11 +6,10 @@ use std::isize;
 use std::iter::FusedIterator;
 use std::mem;
 use std::panic::{RefUnwindSafe, UnwindSafe};
+use std::process;
 use std::sync::Arc;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::time::{Duration, Instant};
-
-use libc;
 
 use flavors;
 use internal::select::{Operation, SelectHandle, Token};
@@ -265,7 +264,7 @@ impl<T> Sender<T> {
         // counter. It's very difficult to recover sensibly from such degenerate scenarios so we
         // just abort when the count becomes very large.
         if old_count > isize::MAX as usize {
-            unsafe { libc::abort() }
+            process::abort();
         }
 
         Sender(chan)

--- a/src/internal/utils.rs
+++ b/src/internal/utils.rs
@@ -2,11 +2,11 @@
 
 use std::cell::Cell;
 use std::num::Wrapping;
+use std::process;
 use std::sync::atomic;
 use std::thread;
 use std::time::Duration;
 
-use libc;
 use rand;
 
 /// A counter that performs exponential backoff in spin loops.
@@ -58,7 +58,7 @@ impl Drop for AbortGuard {
             line!(),
             column!(),
         );
-        unsafe { libc::abort() }
+        process::abort();
     }
 }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -280,7 +280,6 @@
 
 extern crate crossbeam_epoch;
 extern crate crossbeam_utils;
-extern crate libc;
 extern crate rand;
 extern crate parking_lot;
 

